File: 1281.subtract-the-product-and-sum-of-digits-of-an-integer.java
/*
 * @lc app=leetcode id=1281 lang=java
 *
 * [1281] Subtract the Product and Sum of Digits of an Integer
 */

// @lc code=start
class Solution {
    public int subtractProductAndSum(int n) {
        int prod = 1;
        int su = 0;
        while (n > 0) {
            int digit = n % 10;
            prod *= digit;
            su += digit;
            n = n / 10;
        }
        return prod - su;
    }
}
// @lc code=end

File: 1281.subtract-the-product-and-sum-of-digits-of-an-integer.py
#
# @lc app=leetcode id=1281 lang=python
#
# [1281] Subtract the Product and Sum of Digits of an Integer
#

# @lc code=start
class Solution(object):
    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """
        prod = 1
        su = 1
        while n > 0:
            d = n % 10
            prod *= d
            su += d
            n //= 10
        return prod - su + 1
        
# @lc code=end

